### PR TITLE
Support option/shift in spacing's scrub

### LINKED ISSUE #668: Change multiple spacing values at the same time using scrub
In the spacing UI user can change values using scrub:

Screenshot 2022-12-20 at 15 11 40

Special rules should apply when user holds OPTION/CONTROL:

Screenshot 2022-12-20 at 15 06 35

### PR DESCRIPTION
**Pull request by rpominov**
## Description

Adds option / shift support to scrub in spacing. Closes #668

## Steps for reproduction

- Try to scrub in spacing while holding SHIFT or OPTION (ALT in Windows)
- Try to press and release SHIFT / OPTION multiple times during scrub
- Try to press SHIFT / OPTION while hovering over a property (other properties will highlight even without scrub)

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

**Review by kof — APPROVED** [review]
Works well, nice!

File: apps/designer/app/designer/features/style-panel/sections/spacing/scrub.tsx [hunk 1]
```diff
@@ -35,64 +53,86 @@ export const useScrub = (props: {
...
+      ["marginTop", "marginRight", "marginBottom", "marginLeft"],
+    ];
+  } else if (modifiers.altKey) {
+    groups = [
+      ["paddingTop", "paddingBottom"],
+      ["paddingRight", "paddingLeft"],
+      ["marginTop", "marginBottom"],
+      ["marginRight", "marginLeft"],
+    ];
+  }
```
Comment (kof) [inline apps/designer/app/designer/features/style-panel/sections/spacing/scrub.tsx:135]: nit: these groups could be static constants outside

File: apps/designer/app/designer/features/style-panel/shared/modifier-keys.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from "react";
+
+export const useModifierKeys = () => {
```
Comment (kof) [inline apps/designer/app/designer/features/style-panel/shared/modifier-keys.ts:3]: Maybe a hit when this is used: used in combination of modifier keys and mouse interactions, e.g. spacing control shift + mouse

File: apps/designer/app/designer/features/style-panel/shared/use-style-data.ts [hunk 3]
```diff
@@ -174,27 +174,31 @@ export const useStyleData = ({
...
-          { ...prevStyle }
-        );
-        updates = [];
-        return nextStyle;
-      });
+      const type = options.isEphemeral ? "preview" : "update";
+      publishUpdates(type, updates);
+
+      if (options.isEphemeral === false) {
+        setBreakpointStyle((prevStyle) => {
```
Comment (kof) [inline apps/designer/app/designer/features/style-panel/shared/use-style-data.ts:185]: we renamed breakpoint style to local style, maybe its not merged yet, @TrySound

Comment (rpominov) [reply apps/designer/app/designer/features/style-panel/shared/use-style-data.ts:185]: Probably not merged. I didn't change this code, only wrapped it into an `if`

**Pull request merged**